server/src/api/api.ts

```diff
@@ -248,11 +305,131 @@ const resolvers: any = {
+            args.newRequest.quantity = 1;
+        }
+
+        // TODO: if item has approvalRequired == false, initial status should be APPROVED as long as the total qty
+        //       of this item the user has requested
+        let initialStatus: RequestStatus = "SUBMITTED";
+        console.log("item", item);
+        if (!item.approvalRequired) {
+            initialStatus = "APPROVED";
+        }
+
+        // return the request object with the item, censoring price/owner for non-admins
+        let newRequest = await DB.from("requests").insert({
+            ...args.newRequest,
+            status: initialStatus
+        })
+            .returning(["request_id", "created_at", "updated_at"]);
+
+        newRequest = newRequest[0];
+        return {
+            request_id: newRequest.request_id,
+            quantity: args.newRequest.quantity,
+            status: initialStatus,
+            item,
+            user,
+            createdAt: localTimestamp(newRequest.created_at),
+            updatedAt: localTimestamp(newRequest.updated_at)
+        };
+    },
+    deleteRequest: async (root, _args, _context): Promise<boolean> => {
+        // @ts-ignore
+        const {args, context} = fixArguments(root, _args, _context);
+
+        if (!context.user.admin) {
+            throw new GraphQLError("You do not have permission to access the deleteRequest endpoint.");
+        }
+
+        const numRowsAffected = await DB.from("requests")
+            .where({
+                request_id: args.id,
+            })
+            .del();
+
+        return numRowsAffected !== 0;
+    },
+    updateRequest: async (root, _args, _context) => {
+        // @ts-ignore
+        const {args, context} = fixArguments(root, _args, _context);
+
+        if (!context.user.admin) {
+            throw new GraphQLError("You do not have permission to access the updateRequest endpoint.");
+        }
+
+        const updateObj: any = {};
+
+        // Not going to validate against maxRequestQty since only admins can change this currently
+
+        const newQuantity = args.updatedRequest.new_quantity;
+        if (newQuantity) {
+            if (newQuantity <= 0) {
+                throw new GraphQLError(`Invalid new requested quantity of ${newQuantity} specified.  The new requested quantity must be >= 1.`);
+            }
+            updateObj.quantity = Math.max(newQuantity, 1);
+        }
+
+        // TODO: status change validation logic
+        if (args.updatedRequest.new_status) {
+            updateObj.status = args.updatedRequest.new_status;
+        }
+
+        if (Object.keys(updateObj).length >= 1) {
+            updateObj.updated_at = new Date();
+
+            const updatedRequest = await DB.from("requests")
+                .where({
+                    request_id: args.updatedRequest.request_id,
+                })
+                .update(updateObj)
+                .returning(["request_id", "quantity", "status", "created_at", "updated_at"]);
```

Comment:
same nit with `created_at`

Reply:
>can we return just updated_at always

Not sure what you mean here.  This is what you get back from the database so I don't think there's a way to always return some columns.

Reply:
I just didn't see the point in returning 'created_at' to the client

Reply:
I don't see the harm in the API being able to return it.  Created at and updated at are fairly standard fields in the API world.